gc/gc-base/src/main/java/org/projectnessie/gc/base/ContentBloomFilter.java

```diff
@@ -54,29 +53,24 @@ public boolean mightContain(Content content) {
   public void merge(ContentBloomFilter filter) {
     if (filter.filter != null) {
       this.filter.putAll(filter.filter);
-      this.wasMerged = true;
     }
   }
 
   public double getExpectedFpp() {
     return filter.expectedFpp();
   }
 
-  /**
-   * A merged bloomfilter might indicate decreased filter quality.
-   *
-   * @return true if other filters were merged into this instance
-   */
-  public boolean wasMerged() {
-    return wasMerged;
-  }
-
   private String getValue(Content content) {
+    // For the contents before global state removal (Nessie version < 0.26.0)
+    // metadataLocation will not be unique.
+    // Hence, to handle both the version's compatibility, use snapshotId + metadataLocation as key.
     switch (content.getType()) {
       case ICEBERG_TABLE:
-        return ICEBERG_TABLE.name() + ((IcebergTable) content).getSnapshotId();
+        IcebergTable icebergTable = (IcebergTable) content;
+        return icebergTable.getSnapshotId() + icebergTable.getMetadataLocation();
```

Comment:
As there is no global state. Same table in other branch may get same snapshot id. So, keeping `content-id + snapshot id` as key will make the content in other branch as live even though it is expired.
In above case tableMetadata will be unique. So, Using that as a key.

Reply:
The purpose of Nessie GC is to mark snapshots as live / identify snapshots that are not live and then expire particular snapshots. The key for that is content-ID + snapshot-ID. I don't understand your point about "there is no global state".

Reply:
same table in different branches can have the same snapshot id allocated for different operations right? In that case, maybe one operation is live and the other is expired. So, we cannot rely on just content-ID + snapshot-ID to identify live or not (because we are merging the bloom filter of all the references into one)

Reply:
RIght. okay.